### DISCUSSION TITLE
500 when trying to queue a test (was working a few months ago)

### DISCUSSION THREAD

**Discussion by NULUSIOS** [alexjustesen/speedtest-tracker#533] [Q&A] [answered]
So I have this installed on my UNRAID since late 2022 and was working fine, did a few tests ok.
I wanted to try it out today, it is updated already to latest version 0.11.12...
When I click to queue a test, immediately get 500 error.

Any ideas?

**Comment by alexjustesen** [comment]
@NULUSIOS if you browse the logs any messages show up?

**Comment by NULUSIOS** [comment]
There is an early error before even going in:
`[2023-04-10 17:08:03] production.ERROR: array_rand(): Argument #1 ($array) must be of type array, string given {"exception":"[object] (TypeError(code: 0): array_rand(): Argument #1 ($array) must be of type array, string given at /var/www/html/app/Jobs/SearchForSpeedtests.php:27)
`
and this (among other things), when I click to queue:
`[2023-04-10 17:15:04] production.ERROR: array_rand(): Argument #1 ($array) must be of type array, string given {"exception":"[object] (TypeError(code: 0): array_rand(): Argument #1 ($array) must be of type array, string given at /var/www/html/app/Jobs/SearchForSpeedtests.php:27)
[stacktrace]`

**Reply by alexjustesen** [reply]
Do you have any custom speedtest servers added?

**Reply by NULUSIOS** [reply]
no I don't think so

**Reply by alexjustesen** [reply]
Try adding one or a couple, there is a bug I think in that particular field that it sometimes thinks there is an array of values that turns out to be empty.

If you add some and remove them again it should fix the isssue.

**Reply by NULUSIOS** [reply]
Can you be more specific? Where do I define a custom server? Edit some config file? In the GUI I don't think I can find this.
Also, does it matter if the server actually exists? (because I don't know of any speedtest server)

**Reply by alexjustesen** [reply, marked as answer]
If you go into general settings you can set the servers here, when you click in the field it'll load the "closest" ones for you. 
![image](https://user-images.githubusercontent.com/1144087/231303584-b8bb8dbd-f189-4c6e-8863-f5770b04eef7.png)

**Reply by NULUSIOS** [reply]
Yes, that seems to do it.
Can I remove again, or the bug will return?

**Reply by alexjustesen** [reply]
You should be able to remind it

**Reply by NULUSIOS** [reply]
Indeed after removing, it still works.
Hope the bug is fixed.
